source/admin_manual/importing_mailboxes.rst

```diff
@@ -55,3 +55,81 @@ If you have only read privileges, you can try using in-memory indexes.
+ * Old recovered mails are imported on top of the newly received mails. The
+   downside here is that mails may now be sorted in a weird order. If the IMAP
+   client shows the mails in the saved order, the new received emails show up
+   as oldest emails. Although this may not be an issue, since many IMAP clients
+   sort the mails by either Date: header or the received timestamp (IMAP
+   INTERNALDATE).
+ * New mails are imported on top of the old recovered mails. This avoids the
+   sorting problems, so it's likely the preferred method. The downside here is
+   that the IMAP/POP3 clients will have to re-download also the newly delivered
+   emails, as well as the old ones. Another issue with this is that some IMAP
+   clients might not show the old recovered mails without manually rebuilding
+   local caches, because the mails become inserted to the beginning of the
+   folders, which isn't allowed by the IMAP protocol.
+
+     * Another thing to keep mind here is that IMAP clients shouldn't see
+       IMAP UIDs pointing to different emails before/after the merge. Otherwise
+       their local cache could point to a different email, which could even
+       cause the user to delete wrong messages. This shouldn't be an issue as
+       long as new mail deliveries and all user access is disabled during the
+       merging. The old mails have the old UIDs, and newly delivered mails
+       would all have higher UIDs (because the UIDNEXT value is not shrunk
+       during index rebuild that clears out the mailbox).
+
+ * New recovered mails are imported under a separate ``Recovered/`` folder,
+   i.e. there will be ``Recovered/INBOX``, ``Recovered/Sent``, etc. The user
+   will need to manually merge the folders. The upside here is that POP3
+   clients won't re-download any mails as duplicates, but otherwise it's not
+   much different from the 1st case.
+
+Example for the 2nd case where mail storage broke down, but a separate index
+storage is ok, and index storage supports snapshots:
+
+ * Snapshot the current index volume at the time of breakage
+ * Make sure ``mail_location`` setting has ``ITERINDEX`` feature enabled, so
+   folder listing is done using the index volume rather than the mail volume.
+ * Mount a new empty mail volume
+
+    * The first time IMAP/POP3 client attempts to access an existing mail,
+      Dovecot rebuilds the indexes for the folder. This makes the folder look
+      empty. The folder structure is preserved, as long as ``ITERINDEX``
+      setting is used.
+
+ * Once the original mail volume is recovered, first disable all user access
+   and all new mail deliveries.
+ * Snapshot again the index volume
+ * Mount the old mail volume to the original mountpoint
+ * Replace the index volume with the first created snapshot. Now the storage
+   looks exactly like it was at the time of breakage.
+ * Mount the new mail volume to some temporary mountpoint
+ * Mount the second index snapshot to some temporary mountpoint
```

Comment:
Some of these lines are terminated with a `.` some aren't, it should be consistent.

And:
"Create another snapshot of the index volume." Otherwise it reads a bit klunky.